fms/utils/serialization.py

```diff
@@ -58,6 +69,99 @@ def list_sources(architecture: str):
     return list(__adapters[architecture].keys())
 
 
+def _legacy_attn_unfused_to_fused_weight_conversion(
+    name: str, orig_sd: Mapping
+) -> Optional[Tuple[str, torch.Tensor]]:
+    """
+    function which converts unfused fms weights to fused fms weights in the case the model was using the older unfused
+    weights (version 0.0.3)
+
+    Args:
+        name: str
+            current name to convert
+        orig_sd: Mapping
+            a mapping from a name to a param, in most cases will be a singleton, however when
+
+    Returns:
+    Optional[Tuple[str, torch.Tensor]]
+        if query/key/value all exist in the given state dict, a tuple of the new fused name as well as weights will be
+        returned from this function
+        if one of query, key, or value exists in state dict (not all together), a FusableWeightsMissingError will be
+        raised with the weights that are missing
```

Comment:
the exception no longer exists, but I would still raise a RuntimeError or ValueError when they're missing, as it might be an issue of the state dict being malformed on disk

Reply:
fixed in latest commit